HW14/task1.py

```diff
@@ -0,0 +1,131 @@
+import json
+import re
+import datetime
+
+filename = 'phoonebook.json'
+
+def validate_phone_number(number):
+#    pattern = r'^(?:\+?38|0)?\d{9,10}$'
+    pattern = '^(?:\+?38)?0\d{9}$'
+    return bool(re.match(pattern, number))
+class CustomException(Exception):
+    def __init__(self, message):
+        self.message = message
+        self.timestamp = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        with open("error.log", "a") as f:
+            f.write(f"Error at {self.timestamp}: {self.message}\n")
+        super().__init__(self.message)
```

Comment:
nit: add 2 blank lines before and after class and one line before and after the function
```suggestion


class CustomException(Exception):
    def __init__(self, message):
        self.message = message
        self.timestamp = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
        with open("error.log", "a") as f:
            f.write(f"Error at {self.timestamp}: {self.message}\n")
        super().__init__(self.message)


```